### DISCUSSION TITLE
How to add analytics label to cloud messaging?

### DISCUSSION THREAD

**Discussion by FionaYam** [invertase/react-native-firebase#5105] [Q&A] [answered]
I would like to add Messages Label in my Firebase Cloud Messaging Reports. Their documentation mentions that should add an analytics label.

https://firebase.google.com/docs/cloud-messaging/understand-delivery?authuser=0#adding-analytics-labels-to-messages

<img width="1278" alt="fcm" src="https://user-images.githubusercontent.com/45822881/113168521-559c4400-9277-11eb-9781-0b221b93f755.png">

How should I do this in this package? I can't find in the documentation.

````
                const localNotification = new firebase.notifications.Notification()
                    .setNotificationId(notification.notificationId)
                    .setTitle(notification.title)
                    .setSubtitle(notification.subtitle)
                    .setBody(notification.body)
                    .setData(notification.data)
                    .setSound('default')
                    .android.setChannelId('channelId')
                    .android.setPriority(firebase.notifications.Android.Priority.High);

                firebase.notifications()
                    .displayNotification(localNotification)
                    .catch(err => console.log(err));

**Comment by mikehardy** [comment, marked as answer]
Hi @FionaYam !

I don't think you have to do anything with regard to react-native-firebase

As they mention in the doc you linked, this is a JSON option that you would set using the firebase-admin-sdk in a cloud function or similar as you *create* FCM message JSON that you send to clients.

The native firebase-ios-sdk/firebase-android-sdk that react-native-firebase wraps on the client side will consume the FCM JSON that you created, and if it contains the correct keys, it should "Just Work(tm)"
